server/base_client.py

```diff
@@ -100,5 +103,9 @@ def send(self, msg):
             msg = msg.encode('utf-8')
         if isinstance(msg, Message):
             msg = msg.as_bytes()
-        self.conn.send(msg)
+        try:
+            self.conn.send(msg)
+        except BrokenPipeError:
+            log.warning('BrokenPipeError - %s' % self.user.name)
```

Comment:
It's better to use comma in logs e.g.
log.warning('BrokenPipeError - %s', self.user.name)
In this case, a sting will not build in case a log shouldn't be written.

Reply:
fixed